src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditor.java

```diff
@@ -29,45 +34,72 @@
 import org.lwjgl.input.Mouse;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
 
 public class GuiPositionEditor extends GuiScreen {
-	private final Position position;
-	private final Position originalPosition;
-	private final int elementWidth;
-	private final int elementHeight;
-	private final Runnable renderCallback;
+	private final ArrayList<Position> positions;
+	private final ArrayList<Position> originalPositions;
+	private final ArrayList<Integer> elementWidths;
+	private final ArrayList<Integer> elementHeights;
+	private final ArrayList<Runnable> renderCallback;
 	private final Runnable positionChangedCallback;
 	private final Runnable closedCallback;
-	private boolean clicked = false;
 	private int grabbedX = 0;
 	private int grabbedY = 0;
+	private int clickedPos = -1;
+
+	public static boolean renderDrill = false;
 
 	private int guiScaleOverride = -1;
 
 	public GuiPositionEditor(
-		Position position, int elementWidth, int elementHeight,
+		LinkedHashMap<TextOverlay, Position> overlayPositions,
 		Runnable renderCallback,
 		Runnable positionChangedCallback,
 		Runnable closedCallback
 	) {
-		this.position = position;
-		this.originalPosition = position.clone();
-		this.elementWidth = elementWidth;
-		this.elementHeight = elementHeight;
-		this.renderCallback = renderCallback;
-		this.positionChangedCallback = positionChangedCallback;
-		this.closedCallback = closedCallback;
-	}
+		ArrayList<Position> pos = new ArrayList<>();
+		ArrayList<Position> ogPos = new ArrayList<>();
+		ArrayList<Runnable> renderCallbac = new ArrayList<>();
+		ArrayList<Integer> width = new ArrayList<>();
+		ArrayList<Integer> height = new ArrayList<>();
+		for (int i = 0; i < overlayPositions.size(); i++) {
+			TextOverlay overlay = new ArrayList<>(overlayPositions.keySet()).get(i);
+			pos.add(overlayPositions.get(overlay));
+			ogPos.add(pos.get(i).clone());
+			width.add((int) overlay.getDummySize().x);
+			height.add((int) overlay.getDummySize().y);
+			if (i < 10) {
+				renderCallbac.add(() -> {
+					overlay.renderDummy();
+					OverlayManager.dontRenderOverlay = overlay.getClass();
+				});
+			}
```

Comment:
what is this all about

Reply:
Not sure what part you mean but I changed the if statement 
the render callback is what renders the text ( :scream: )